Fix typo in a parameter name for open llama model

aaalexlit opened the pull request:
# What does this PR do?

Renames a parameter `use_memorry_efficient_attention` to `use_memory_efficient_attention`

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->


## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts
- speech models: @sanchit-gandhi
- graph models: @clefourrier

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @sgugger

Integrations:

- deepspeed: HF Trainer: @stas00, Accelerate: @pacman100
- ray/raytune: @richardliaw, @amogkam

Documentation: @sgugger, @stevhliu and @MKhalusova

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: @sgugger
- TensorFlow: @Rocketknight1

 -->

aaalexlit committed:
Update modeling_open_llama.py

Fix typo in `use_memorry_efficient_attention` parameter name

aaalexlit committed:
Update configuration_open_llama.py

Fix typo in `use_memorry_efficient_attention` parameter name

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_23637). All of your documentation changes will be reflected on that endpoint.

amyeroberts commented:
Hi @aaalexlit, thanks for opening this PR! 

As the model was part of the most recent release, we'll need to ensure that these changes are backwards compatible with the previous configurations. What I would suggest is popping the previous argument `use_memorry_efficient_attention`  from the config kwargs during initialization, and using that value if passed in, otherwise defaulting to `use_memory_efficient_attention`.

aaalexlit committed:
Update configuration_open_llama.py

Take care of backwards compatibility ensuring that the previous parameter name is taken into account if used

aaalexlit commented:
Thanks for pointing that out, @amyeroberts! Makes all the senses in the world.
I hope I interpreted your recommendation correctly

aaalexlit committed:
Update configuration_open_llama.py

format to adjust the line length

aaalexlit committed:
Update configuration_open_llama.py

proper code formatting using `make fixup`

sgugger commented on src/transformers/models/open_llama/configuration_open_llama.py:
```diff
@@ -116,7 +116,10 @@ def __init__(
         self.use_cache = use_cache
-        self.use_memorry_efficient_attention = use_memorry_efficient_attention
+        if "use_memorry_efficient_attention" in kwargs:
+            self.use_memory_efficient_attention = kwargs["use_memorry_efficient_attention"]
```
```suggestion
            self.use_memory_efficient_attention = kwargs.pop("use_memorry_efficient_attention")
```
If you leave it here, it will still set a `self.use_memorry_efficient_attention`

sgugger reviewed (approved):
Thanks for the fix!

aaalexlit committed:
Update configuration_open_llama.py

pop the argument not to let it be set later down the line

amyeroberts reviewed (approved):
Thanks for fixing and iterating!
